src/motors.c

```diff
@@ -56,6 +150,33 @@ extern void init_motors()
 /* Robert's extern/visible functions go below here: */
 extern void motor_set_speed_percent(motors_t motor_x, uint8_t speed, direction_t dir)
 {
+	if(dir == Reverse) speed *= -1;
+	switch(motor_x){
+		case Motor1:
+			TIM_SetCompare1(TIM2, speed*10);
+			break;
+		case Motor2:
+			TIM_SetCompare4(TIM2, speed*10);
+			break;
+		case Motor3:
+			TIM_SetCompare3(TIM1, speed*10);
+			break;
+		case Motor4:
+			TIM_SetCompare1(TIM1, speed*10);
+			break;
+		case Motor5:
+			TIM_SetCompare4(TIM4, speed*10);
+			break;
+		case Motor6:
+			TIM_SetCompare1(TIM4, speed*10);
+			break;
+		case Motor7:
+			TIM_SetCompare1(TIM3, speed*10);
+			break;
+		case Motor8:
+			TIM_SetCompare3(TIM3, speed*10);
```

Comment:
uint8_t max number is 2^8=255, the max input speed we'd receive is 100, 100*10=1000 > max number for uint8_t. Not entirely sure what the input parameters to that function is, but if there may be an overflow we'd have to watch out for.

Reply:
The max compare value is 1000. It's the max value of the capture compare register. Still not sure why it is timed by 10? If you ask me this should take in raw speed, we can have another function or use the FSM to convert 0-255 to what it is in 10000.